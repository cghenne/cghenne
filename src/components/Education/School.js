// @flow
import React from 'react';
import type { SchoolT } from './constants';
import './Education.scss';

type Props = SchoolT;

const School = ({ graduation_year, school, location, description }: Props) => {
  return (
    <div className="school">
      <div className="year">{graduation_year}</div>
      <div className="details">
        <h4>{school}</h4>
        <h5>{location}</h5>
        <p>{description}</p>
      </div>
    </div>
  );
};

School.displayName = 'School';

export default School;
