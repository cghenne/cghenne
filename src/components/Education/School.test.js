import React from 'react';
import { shallow } from 'enzyme';
import School from './School';

const fakeSchool = {
  graduation_year: 2022,
  school: 'University of Toronto',
  location: 'Toronto, CA',
  description: `Master of Science in Applied Computing (MScAC)`
};

describe('School', () => {
  const wrapper = shallow(<School {...fakeSchool} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the school in a scholl class', () => {
    expect(
      wrapper
        .find('div')
        .first()
        .prop('className')
    ).toEqual('school');
  });

  it('should render the graducation_date dates in a year class', () => {
    expect(wrapper.find('.year').text()).toEqual(
      fakeSchool.graduation_year.toString()
    );
  });

  it('should render the details of the school in a details class', () => {
    const details = wrapper.find('.details');
    expect(details.exists()).toEqual(true);
    expect(details.find('h4').text()).toEqual(fakeSchool.school);
    expect(details.find('h5').text()).toEqual(fakeSchool.location);
    expect(details.find('p').text()).toEqual(fakeSchool.description);
  });
});
