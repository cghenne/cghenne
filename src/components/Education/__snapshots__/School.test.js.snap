// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`School should render without crashing 1`] = `
<div
  className="school"
>
  <div
    className="year"
  >
    2022
  </div>
  <div
    className="details"
  >
    <h4>
      University of Toronto
    </h4>
    <h5>
      Toronto, CA
    </h5>
    <p>
      Master of Science in Applied Computing (MScAC)
    </p>
  </div>
</div>
`;
